tweag/nixpkgs-graph
# The first graph

## What this PR does

This generates the first graph via [networkx](https://networkx.org/) and classifies the nodes by the package they belong to.

### Principal changes:
We modified the json data generated by `default.nix`, removed the `outPath` attribute so that more packages could be evaluated, and used [pandas](https://pandas.pydata.org/) to read and process the json file, and `networkx` to generate the first graph.

### Compared with the version \#10
- The `outPath` is no longer evaluated.
- This added `pythonPackages` to the whitelist.
- This created the `nixpkgs-graph` python project to generate the graph.
- This adds the virtual environment `.venv` needed for python project.
- This adds the `shell.nix` file for providing `python3`.
- This will generate the `nodes.csv` file and the `first_graph.png` image in `rawdata/` folder.
- This modified the `build.sh` file and `README.md` to accommodate the new approach.
- This adds the MIT License.
## How to test this PR
- Install nix package manager if this is not done:
  ```sh
  $ curl -L https://nixos.org/nix/install | sh
  ``` 

- Excute the following command to generate the database and the graph.
  ```sh
  $ /bin/bash build.sh
  ``` 

- Observe the presence of `nodes.csv` and `first_graph.png` files in the `rawdata/` folder.
